internal/app/machined/pkg/runtime/v1alpha1/platform/metal/metal.go

```diff
@@ -162,7 +162,7 @@ func (m *Metal) KernelArgs(arch string) procfs.Parameters {
 	switch arch {
 	case "amd64":
 		return []*procfs.Parameter{
-			procfs.NewParameter("console").Append("ttyS0").Append("tty0"),
```

Comment:
we need to make it dependent on Talos version (`quirks`) - we want to make images for Talos < 1.8 still append `ttyS0`.

Reply:
we also need to figure out compatibility on upgrades - if the installed Talos runs with `console=ttyS0`, we should preserve that.